### PR TITLE
Basic link formatting support; component tests

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
There doesn't seem a reasonable approach that can avoid `@html` in the Svelte. Perhaps with significant changes to svelte-i18n we could render it all to DOM or some other intermediate representation but that seems a lot of effort for an odd API.

### CODE REVIEW

File: src/setup_tests.ts [hunk 1]
```diff
@@ -4,6 +4,13 @@
  * SPDX-License-Identifier: MIT
  */
 
+import * as matchers from 'vitest-dom/matchers';
+import { expect } from 'vitest';
+expect.extend(matchers);
+
+// or:
```
Comment (microbit-matt-hillsdon) [inline src/setup_tests.ts:11]: Hah, I blindly copied both options!

```suggestion
```

File: src/components/HtmlFormattedMessage.test.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    render(HtmlFormattedMessage, { id: 'simple' });
+    const dom = screen.getByText('An example');
+    expect(dom).toMatchInlineSnapshot(`
+      <span>
+        An example
+      </span>
+    `);
+  });
+
+  it('renderes a link', () => {
```
Comment (microbit-matt-hillsdon) [inline src/components/HtmlFormattedMessage.test.ts:52]:
```suggestion
  it('renders a link', () => {
```

File: src/components/HtmlFormattedMessage.test.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  t: {
+    subscribe(cb: any) {
+      cb(tValue);
+      return () => {};
+    },
+  },
+}));
+
+describe('HtmlFormattedMessage', () => {
+  it('renderes simple example', () => {
```
Comment (microbit-grace) [inline src/components/HtmlFormattedMessage.test.ts:42]:
```suggestion
  it('renders simple example', () => {
```

File: src/setup_tests.ts [hunk 4]
```diff
@@ -4,6 +4,9 @@
  * SPDX-License-Identifier: MIT
  */
 
+import * as matchers from 'vitest-dom/matchers';
+import 'vitest-dom/extend-expect';
+
```
Comment (microbit-grace) [inline src/setup_tests.ts:9]: I don't think these are used?

File: src/components/connection-prompt/usb/manual/ManualInstallTutorial.svelte [hunk 5]
```diff
@@ -9,16 +9,21 @@
   import { onMount } from 'svelte';
   import StandardButton from '../../../StandardButton.svelte';
   import Microbits from '../../../../script/microbit-interfacing/Microbits';
-  import ImageSkeleton from '../../../skeletonloading/ImageSkeleton.svelte';
   import Bowser from 'bowser';
   import transferFirmwareMacOSImage from '../../../../imgs/transfer_firmware_macos.gif';
   import transferFirmwareChromeOSImage from '../../../../imgs/transfer_firmware_chromeos.gif';
   import transferFirmwareWindowsImage from '../../../../imgs/transfer_firmware_windows.gif';
   import DialogHeading from '../../../DialogHeading.svelte';
+  import EscapedChunksFormattedMessage, {
```
Comment (microbit-grace) [inline src/components/connection-prompt/usb/manual/ManualInstallTutorial.svelte:17]: I don't think `EscapedChunksFormattedMessage` is used or defined?

**Review by microbit-grace — COMMENTED** [review]
Looks good, left a few minor comments

File: src/setup_tests.ts (hunk 4 above)
Comment (microbit-matt-hillsdon) [reply src/setup_tests.ts]: I think we want the second one as step 4 of https://testing-library.com/docs/svelte-testing-library/setup

I'd accidentally done both of the options for that step and then not quite cleaned up afterwards. We're not actually using them yet but seemed prudent to follow the steps. Tidied in 4472aa4a2838bf7c217992c1bd00e5729f4e6a6e

File: src/components/connection-prompt/usb/manual/ManualInstallTutorial.svelte (hunk 5 above)
Comment (microbit-matt-hillsdon) [reply src/components/connection-prompt/usb/manual/ManualInstallTutorial.svelte]: 48f6c50b09be33c24a00448ba8941dbdd6b71d09

**Review by microbit-grace — APPROVED** [review]
LGTM

**Review by microbit-robert — APPROVED** [review]
LGTM.

**Comment by microbit-matt-hillsdon** [comment]
Fixed trivial import conflict, will merge once rebuilt.

**Pull request merged**